enclave/core/malloc.c

```diff
@@ -132,147 +50,45 @@ void* oe_calloc(size_t nmemb, size_t size)
 
 void* oe_realloc(void* ptr, size_t size)
 {
-    void* p = REALLOC(ptr, size);
+    void* p = oe_allocator_realloc(ptr, size);
 
     if (!p && size)
     {
-        errno = ENOMEM;
-
         if (_failure_callback)
             _failure_callback(__FILE__, __LINE__, __FUNCTION__, size);
     }
 
     return p;
 }
 
-int oe_posix_memalign(void** memptr, size_t alignment, size_t size)
+void* oe_memalign(size_t alignment, size_t size)
 {
-    int rc = POSIX_MEMALIGN(memptr, alignment, size);
-
-    if (rc != 0 && size)
-    {
-        errno = ENOMEM;
+    void* ptr = NULL;
 
-        if (_failure_callback)
-            _failure_callback(__FILE__, __LINE__, __FUNCTION__, size);
-    }
+    // The only difference between posix_memalign and the obsolete memalign is
+    // that posix_memalign requires alignment to be a multiple of sizeof(void*).
+    // Adjust the alignment if needed.
+    if (alignment < sizeof(void*))
```

Comment:
Same comment as above

Reply:
Made it more readable using round to multiple.